### ISSUE TITLE
Suggestion: XMR private viewkey export by default

### ISSUE THREAD

**Issue by dnaleor** [LedgerHQ/ledger-app-monero#27]
To enhance the UX , I propose the following:

When the **first time** the Ledger asks to export the viewkey, the ledger could instead ask:

> "Export viewkey by default?"

YES => The viewkey will always be exported, no need to push the button each time. There should be an option in the Monero app to turn the default exporting off.
NO => the ledger won't ask this question again, but it'll ask the following:

> "Export viewkey during this session?"

YES => The viewkey will be exported by default until the ledger is disconnected. 
NO => the ledger could ask the following:

> "One time viewkey export?"

**Comment by cslashm** [comment]
I understand the need. I suggest  a new entry in setting to set the behavior.

**Comment by mmbyday** [comment]
+1 This is extra friction and often confusing for the beginner/average monero user.

**Comment by cslashm** [comment]
The double press is a sideeffect of client code at creation. I will try to fix it for 0.14 if not too buzy on other stuff

**Comment by snaggen** [comment]
Having a setting would improve usability a lot... I tend to forget that I need to export the viewkey, so I just plugin my ledger and open the wallet and then it looks like nothing works.... until I 30 seconds later looks around trying to figure out what is wrong and sees the text on the Ledger. This happens to me more often than it should... so having a default behaviour in a setting would be nice.

**Comment by cslashm** [comment]
planned in 1.4.0

**Comment by Mansarde** [comment]
@dnaleor 
What is the difference between session and one-time export?

I understand the session export like this:
- I connect the Ledger to my PC and allow session export.
- I then open and close the Monero GUI multiple times without being asked again by the Ledger, until I re-connect the Ledger.

Would the one-time export work like this?
- I connect the Ledger to my PC and allow one-time export.
- Every time (within the same Ledger session) I re-open the Monero GUI the Ledger asks me to export again?

**Comment by dnaleor** [comment]
@Mansarde yeah something like that. Although you may simplify and just only use "session export" or only the "one time export". The most important change would be a flag to export by default

**Comment by skorokithakis** [comment]
Agreed about the necessity of this feature. I'd go one further and have wallets optionally store the view key on the device, and thus not require the Ledger to be connected unless it's needed for sending or (more rarely) for getting more subaddresses.
